hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -183,69 +178,64 @@ public HoodieMergeOnReadTable(HoodieWriteConfig config, HoodieTableMetaClient me
                       .deleteCleanedFiles(partitionPath, Arrays.asList(commit));
 
                   // append rollback blocks for updates
-                  commitMetadata.getPartitionToWriteStats().get(partitionPath).stream()
-                      .filter(wStat -> wStat.getPrevCommit() != HoodieWriteStat.NULL_COMMIT)
-                      .forEach(wStat -> {
-                        HoodieLogFormat.Writer writer = null;
-                        try {
-                          writer = HoodieLogFormat.newWriterBuilder()
-                              .onParentPath(
-                                  new Path(this.getMetaClient().getBasePath(), partitionPath))
-                              .withFileId(wStat.getFileId()).overBaseCommit(wStat.getPrevCommit())
-                              .withFs(getMetaClient().getFs())
-                              .withFileExtension(HoodieLogFile.DELTA_EXTENSION).build();
-                          Long numRollbackBlocks = 0L;
-                          // generate metadata
-                          Map<HoodieLogBlock.LogMetadataType, String> metadata = Maps.newHashMap();
-                          metadata.put(HoodieLogBlock.LogMetadataType.INSTANT_TIME,
-                              metaClient.getActiveTimeline().lastInstant().get().getTimestamp());
-                          metadata.put(HoodieLogBlock.LogMetadataType.TARGET_INSTANT_TIME, commit);
-                          // if update belongs to an existing log file
-                          writer.appendBlock(new HoodieCommandBlock(
-                              HoodieCommandBlock.HoodieCommandBlockTypeEnum.ROLLBACK_PREVIOUS_BLOCK,
-                              metadata));
-                          numRollbackBlocks++;
-                          if (wStat.getNumDeletes() > 0) {
-                            writer.appendBlock(new HoodieCommandBlock(
+                  if(commitMetadata.getPartitionToWriteStats().containsKey(partitionPath)) {
+                    commitMetadata.getPartitionToWriteStats().get(partitionPath).stream()
+                        .filter(wStat -> {
+                          return wStat != null && wStat.getPrevCommit() != HoodieWriteStat.NULL_COMMIT
```

Comment:
this sort of assumes that inserts don't ever go into logs right? does anything break by not checking `getPrevCommit()` ?

Reply:
okay its mostly a base commit.. This could change down the line . I guess we will need to hunt this down then, after redefining clear semantics for this.. ok for now.